### DISCUSSION TITLE
Dynamic highlighter while typing

### DISCUSSION THREAD

**Discussion by LuizPelegrini** [ueberdosis/tiptap#1812] [General Questions] [answered]
Hi there,

I am trying to create a tiptap feature that while the user is typing, it applies a custom style (change the text color or background color) until certain length of the text (suppose 10 chars). After that length, the text is displayed normally.

I created a custom mark
```js
const DynamicHighlighter = Mark.create({
  name: "dynamicHighlighter",

  defaultOptions: {
    HTMLAttributes: {
      class: 'dynamic-highlighter'
    }
  },

  parseHTML(){
    return [
      {
        tag: 'span',
        getAttrs: elem => elem.classList.contains('dynamic-highlighter')
      }
    ];
  },

  renderHTML({ HTMLAttributes }){
    return ['span', mergeAttributes(this.options.HTMLAttributes, HTMLAttributes), 0];
  }
});
```

And this is how I use it in my component
```vue
<script>

...

  mounted() {
    this.editor = new Editor({
      extensions: [
        StarterKit,
        DynamicHighlighter
      ],

      content: this.value,

      onUpdate: ({ editor }) => {
        // checking if the limit has been reached...
        const end = editor.state.doc.textContent.length + 1;
        if(editor.state.doc.textContent.length <= 10){
          // create my custom mark and wrap in a <span class="dynamic-highlighter">
          const mark = editor.schema.marks.dynamicHighlighter.create();
          const transaction = editor.state.tr;
          transaction.addMark(1, end, mark);
          editor.view.dispatch(transaction);
        }else{
          const transaction = editor.state.tr;
          transaction.removeMark(11, end, editor.schema.marks.dynamicHighlighter);
          editor.view.dispatch(transaction);
        }

        // HTML
        this.$emit("input", editor.getHTML());
      },
     });
  }
</script>
 
<style>
.ProseMirror span.dynamic-highlighter{
  color: rgb(219, 219, 219);
  background-color: blue;
  padding: 2px;
}
</style>
```

Everything works fine if I start typing from the first line

![image](https://user-images.githubusercontent.com/10824374/131130903-1c725548-cc85-4952-876f-8ed612262aba.png)

the problem comes if I start typing on a new line (after pressing enter)
![image](https://user-images.githubusercontent.com/10824374/131131222-405e56e4-f309-4a2d-9837-000c99e05dfa.png)

I am trying to accomplish this task for quite some time. If there is an easier way on how to accomplish this with tiptap, I'd be more than happy to know it, as I have a feeling the way I'm doing is quite hacky and inefficient.

**Comment by philippkuehn** [comment, marked as answer]
You could also solve this with decorations. Read about it in the [ProseMirror guide](https://prosemirror.net/docs/guide/). The [linter experiment](https://www.tiptap.dev/experiments/linter/) is maybe a good reference.

Another way is to dynamically add a class to the editor based on its content:

```js
new Plugin({
  props: {
    attributes: state => {
      if (state.doc.textContent.length <= 10) {
        return {
          class: 'your-custom-class',
        }
      }
    },
  },
})

```
